drivers/focuser/ieaffocus.cpp

```diff
@@ -0,0 +1,709 @@
+    Lesser General Public License for more details.
+
+    You should have received a copy of the GNU Lesser General Public
+    License along with this library; if not, write to the Free Software
+    Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301  USA
+
+*/
+
+#include "ieaffocus.h"
+#include "indicom.h"
+
+#include <stdio.h>
+#include <termios.h>
+#include <string.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <math.h>
+#include <memory>
+
+#define iEAFFOCUS_TIMEOUT 4
+
+#define POLLMS_OVERRIDE  1500
+
+std::unique_ptr<iEAFFocus> ieafFocus(new iEAFFocus());
+
+iEAFFocus::iEAFFocus()
+{
+    FI::SetCapability(FOCUSER_CAN_ABS_MOVE | FOCUSER_CAN_REL_MOVE | FOCUSER_CAN_ABORT|FOCUSER_CAN_REVERSE);
+    lastPos = 0;
+}
+
+iEAFFocus::~iEAFFocus()
+{
+}
+
+bool iEAFFocus::initProperties()
+{
+    INDI::Focuser::initProperties();
+    maxpos    = 99999;
+    // SetZero
+//    IUFillSwitch(&SetZeroS[0], "SETZERO", "Set Current Position to 0", ISS_OFF);
+//    IUFillSwitchVector(&SetZeroSP, SetZeroS, 1, getDeviceName(), "Zero Position", "", OPTIONS_TAB, IP_RW, ISR_1OFMANY, 0,
+//                       IPS_IDLE);
+      SetZeroSP[0].fill("SETZERO", "Set Current Position to 0", ISS_OFF);
+      SetZeroSP.fill(getDeviceName(),"Zero Position", "", OPTIONS_TAB, IP_RW,ISR_1OFMANY, 0,IPS_IDLE);
+
+
+
+    // Maximum Travel
+//    IUFillNumber(&MaxPosN[0], "MAXPOS", "Maximum Out Position", "%5.0f", 1., 99999., 0, 0);
+//    IUFillNumberVector(&MaxPosNP, MaxPosN, 1, getDeviceName(), "FOCUS_MAXPOS", "Position", OPTIONS_TAB, IP_RW, 0, IPS_IDLE );
+
+
+    MaxPositionNP[0].fill("MAXPOSITION", "Maximum position", "%5.0f", 1., 99999., 0., 99999.);
+    MaxPositionNP.fill(getDeviceName(), "FOCUS_MAXPOSITION", "Max. Position",
+                       OPTIONS_TAB, IP_RW, 0, IPS_IDLE);
+
+
+    /* Focuser temperature */
+//    IUFillNumber(&TemperatureN[0], "TEMPERATURE", "Celsius", "%2.2f", 0, 50., 0., 50.);
+//    IUFillNumberVector(&TemperatureNP, TemperatureN, 1, getDeviceName(), "FOCUS_TEMPERATURE", "Temperature",
+//                       MAIN_CONTROL_TAB, IP_RO, 0, IPS_IDLE);
+
+    TemperatureNP[0].fill("TEMPERATURE", "Celsius", "%6.2f", -50., 70., 0., 0.);
+    TemperatureNP.fill(getDeviceName(), "FOCUS_TEMPERATURE", "Temperature",
+                       MAIN_CONTROL_TAB, IP_RO, 0, IPS_IDLE);
+
+
+
+    /* Relative and absolute movement */
+    FocusRelPosN[0].min = 0.;
+    FocusRelPosN[0].max = 5000.;
+    FocusRelPosN[0].value = 0.;
+    FocusRelPosN[0].step = 10.;
+
+    FocusAbsPosN[0].min = 0.;
+    FocusAbsPosN[0].max = 99999.;
+    FocusAbsPosN[0].value = 0.;
+    FocusAbsPosN[0].step = 10.;
+
+    addDebugControl();
+
+    return true;
+
+}
+
+bool iEAFFocus::updateProperties()
+{
+    INDI::Focuser::updateProperties();
+    if (isConnected())
+    {
+	defineProperty(TemperatureNP);
+  //      defineProperty(&MaxPosNP);
+	defineProperty(MaxPositionNP);
+        defineProperty(SetZeroSP);
```

Comment:
This can be done via the SYNC operation, no need to define it here. Can you arbitrary since focuser to any position (including zero)?